jobs/haproxy/templates/haproxy_wrapper.erb

```diff
@@ -105,6 +105,7 @@ haproxy -f ${CONFIG} -W -p ${PID_FILE} <%= flags.join(' ') %>
 
 <%- if !run_in_foreground -%>
 # Since HAProxy requires Daemonized mode for its `nbproc` multi-process feature
-# to work properly, and BPM requires the process to stay running, sleep infinitely here
-sleep infinity
+# to work properly, and BPM requires the process to stay running, wait for the 
+# HAProxy main process to terminate
+while kill -0 $(cat ${PID_FILE}) 2>/dev/null; do sleep 1; done;
```

Comment:
so this basically says: "check every second if we can send signals to the haproxy pid." `kill` will throw an error if the pid is gone so that would break the look. right? 
I think the same could be achieved by simply calling `pgrep` or maybe even `ps`?

Reply:
Yes exactly.
You could get the same with `pgrep` (if it's installed) and ps (if it's not on `busybox`). Kill always works and is there.